lib/exceptions.rb

```diff
@@ -28,7 +28,12 @@ class IncompatibleHost < StandardError; end
     class MachineryError < StandardError; end
 
     class UnknownScope < MachineryError; end
-    class UnknownOs < MachineryError; end
+    class UnknownOs < MachineryError
+      def to_s
+        "Unable to determine the operating system. There does not seem to be an" \
```

Comment:
Should this be `a /etc/os-release` file? And it's not really correct because we also look at `/etc/SuSE-release` and `/etc/redhat-release`.